### PR TITLE
Add Problem #41: Tanh & Soft-Capping

### PR DESCRIPTION
**Pull request by ThierryHJ**
## Summary
Adds a new Easy-difficulty problem where users implement tanh activation using three parts:

- **Forward (my_tanh)** — compute tanh from scratch using `exp`, numerically stable via `2/(1+exp(-2x))-1`
- **Manual Backward (tanh_backward)** — compute gradient using `1 - tanh²(x)`, note it depends on the *output* not the input
- **Soft-Capping (soft_cap_logits)** — `cap · tanh(logits/cap)`, a technique used in Gemma 2 and Griffin to bound logits

## Motivation
- Tanh is a classic activation function commonly asked in ML interviews
- Implementing it from scratch tests understanding of exponentials, numerical stability, and the chain rule
- Logit soft-capping is a modern technique (Gemma 2, Griffin) that applies tanh in a practical setting — good bonus context for interviews
- Complements existing activation exercises: ReLU (#1) and GELU (#19)

## Files Changed
| File | Status | Description |
|------|--------|-------------|
| `torch_judge/tasks/tanh.py` | NEW | Task definition with 6 test cases |
| `templates/41_tanh.ipynb` | NEW | Template notebook (blank stubs for 3 functions) |
| `solutions/41_tanh_solution.ipynb` | NEW | Reference solution (numerically stable) |
| `README.md` | Modified | Added row #41 to Fundamentals table, updated badge 40→41 |

## Test Cases
| # | Test Name | What It Verifies |
|---|-----------|-----------------|
| 1 | Matches torch.tanh | Output matches reference with atol=1e-5 |
| 2 | tanh(0) = 0 and bounded output | Boundary behavior, output in (-1, 1) for ±100 |
| 3 | Shape preservation | Multi-dim tensor shape unchanged |
| 4 | Gradient flow | backward() succeeds, grad shape correct |
| 5 | Manual backward (tanh_backward) | Manual grad matches autograd |
| 6 | Soft-capping bounds logits | Output bounded within (-cap, cap), matches reference |

## Conventions Followed
- Auto-discovery via `_registry.py` — no manual registration needed
- Standard notebook format (markdown → import → implementation → debug → submit)
- Colab badge on both template and solution notebooks
- `{fn}` placeholder in all test code
- `torch.allclose()` for numerical comparison with appropriate tolerances
- Difficulty emoji consistent with existing Easy problems (🟢)

## How to Test
```bash
# From repo root
python3 -c "
from torch_judge.tasks._registry import TASKS
print('tanh' in TASKS)  # True
"
```
Open `solutions/41_tanh_solution.ipynb` in JupyterLab and run all cells — the final `check('tanh')` should show 6/6 tests passed.

### CODE REVIEW

**Comment by duoan** [comment]
Hi @ThierryHJ , thanks for contribution. 

This is a fundamental problem of math which would not align the repo goal.
